test/test_string_map.cpp

```diff
@@ -869,6 +869,95 @@ TEST(test_string_map, set) {
   }
 }
 
+TEST(test_string_map, key_exists) {
```

Comment:
to the reviewer: `key_exists(n)` was added in order to distinguish between failed key retrieval and a key not existing: https://github.com/ros2/rcutils/pull/57/commits/70e7a2c9076f22a16c0d94f4bebe0de5cc1ab27f#diff-f909d0b1a298531f76132943dc55f86aL167